web/views.py

```
@@ -3896,3 +3983,138 @@ def streak_detail(request):
+    """
+    Track when a user shares a course on social media.
+    This is called via AJAX when a user clicks on a social share button.
+    
+    Args:
+        request: HTTP request
+        course_slug: The slug of the course being shared
+        
+    Returns:
+        JsonResponse with the status of the share tracking
+    """
+    from .social_share import create_social_share_discount, notify_user_about_discount
```

Ruff (0.8.2)

Redefinition of unused `create_social_share_discount` from line 124

Remove definition: `create_social_share_discount`

(F811)